packages/dcos-history/extra/history/server.py

```diff
@@ -78,16 +80,15 @@ def _buffer_response_(name):
 def _response_(content):
     return Response(response=content, content_type="application/json", headers=headers_cb())
 
-
-def start():
+if __name__ == '__main__':
     global state_buffer
     logging.basicConfig(format='[%(levelname)s:%(asctime)s] %(message)s', level='INFO')
-
     compress.init_app(app)
-
     if 'HISTORY_BUFFER_DIR' not in os.environ:
         sys.exit('HISTORY_BUFFER_DIR must be set!')
-
     state_buffer = BufferCollection(os.environ['HISTORY_BUFFER_DIR'])
     BufferUpdater(state_buffer, headers_cb).run()
```

Comment:
The BufferUpdater will never be called as implemented. If we are telling gunicorn to import and run the WSGI app for us, then we need to include starting the BufferUpdater when we instantiate the WSGI app. This would be a somewhat confusing approach, so maybe consider doing it like this: http://docs.gunicorn.org/en/latest/custom.html so you can easily call out the starting of the updater thread whenever you start the server